UEBPrep/spec/controllers/playlists_controller_spec.rb

```diff
@@ -1,159 +1,27 @@
-# Message expectations are only used when there is no simpler way to specify
-# that an instance is receiving a specific message.
-
 RSpec.describe PlaylistsController, type: :controller do
-
-  # This should return the minimal set of attributes required to create a valid
-  # Playlist. As you add validations to Playlist, be sure to
-  # adjust the attributes here as well.
-  let(:valid_attributes) {
-    skip("Add a hash of attributes valid for your model")
-  }
-
-  let(:invalid_attributes) {
-    skip("Add a hash of attributes invalid for your model")
-  }
-
-  # This should return the minimal set of values that should be in the session
-  # in order to pass any filters (e.g. authentication) defined in
-  # PlaylistsController. Be sure to keep this updated too.
-  let(:valid_session) { {} }
-
-  describe "GET #index" do
-    it "assigns all playlists as @playlists" do
-      playlist = Playlist.create! valid_attributes
-      get :index, {}, valid_session
-      expect(assigns(:playlists)).to eq([playlist])
-    end
-  end
-
-  describe "GET #show" do
-    it "assigns the requested playlist as @playlist" do
-      playlist = Playlist.create! valid_attributes
-      get :show, {:id => playlist.to_param}, valid_session
-      expect(assigns(:playlist)).to eq(playlist)
-    end
-  end
-
-  describe "GET #new" do
-    it "assigns a new playlist as @playlist" do
-      get :new, {}, valid_session
-      expect(assigns(:playlist)).to be_a_new(Playlist)
-    end
-  end
-
-  describe "GET #edit" do
-    it "assigns the requested playlist as @playlist" do
-      playlist = Playlist.create! valid_attributes
-      get :edit, {:id => playlist.to_param}, valid_session
-      expect(assigns(:playlist)).to eq(playlist)
-    end
-  end
-
-  describe "POST #create" do
-    context "with valid params" do
-      it "creates a new Playlist" do
-        expect {
-          post :create, {:playlist => valid_attributes}, valid_session
-        }.to change(Playlist, :count).by(1)
-      end
-
-      it "assigns a newly created playlist as @playlist" do
-        post :create, {:playlist => valid_attributes}, valid_session
-        expect(assigns(:playlist)).to be_a(Playlist)
-        expect(assigns(:playlist)).to be_persisted
-      end
-
-      it "redirects to the created playlist" do
-        post :create, {:playlist => valid_attributes}, valid_session
-        expect(response).to redirect_to(Playlist.last)
-      end
-    end
-
-    context "with invalid params" do
-      it "assigns a newly created but unsaved playlist as @playlist" do
-        post :create, {:playlist => invalid_attributes}, valid_session
-        expect(assigns(:playlist)).to be_a_new(Playlist)
-      end
-
-      it "re-renders the 'new' template" do
-        post :create, {:playlist => invalid_attributes}, valid_session
-        expect(response).to render_template("new")
-      end
+  context 'with a signed in user' do
```

Comment:
Good example of how to write specs as you add little pieces of functionality to the app.